torch/CMakeLists.txt

```diff
@@ -194,12 +194,6 @@ if (USE_CUDNN)
     list(INSERT 0 TORCH_PYTHON_INCLUDE_DIRECTORIES ${CUDNN_INCLUDE_DIR})
 endif()
 
-if (USE_MIOPEN)
```

Comment:
these compile options are still needed, replace `USE_MIOPEN` with `USE_ROCM`

Reply:
As discussed in #22227, things in this if-clause have never been invoked. In fact, experiment in #22345 shows that setting `USE_MIOPEN=USE_ROCM` fails ROCm builds. I would see this block as something that was forgotten to be removed.

Reply:
Hmm, the CI showed the same error. I've made changes according to your suggestion. Let's wait for the CI...

Reply:
Oh I know what's going on - it's `CAFFE2_USE_CUDNN` gets hipified to `CAFFE2_USE_MIOPEN`

Reply:
I modified #22345 and confirmed that this if block is never executed for ROCm build. ("MIOpen flags are enabled!!!", which indicates the execution of this if block, does not show up in the CI output.)

Reply:
I also removed a previous deletion of `CAFFE2_USE_MIOPEN`, which broke the build, just as you said.